File: permutation.js
function permutation(str1, str2) {
  letters = {}

  for (var i = 0; i < str1.length; i++) {
    if (letters[str1[i]] === 1) {
      letters[str1[i]]++
    } else {
      letters[str1[i]] = 1
    }
  }

  for (var i = 0; i < str2.length; i++) {
    if ( isNaN(letters[str2[i]]) || letters[str2[i]] < 0) {
      return false
    } else {
      letters[str2[i]]--
    }
  }
  return true
}

str1 = 'robin'
str2 = 'nibor'

console.log(permutation(str1,str2))

File: permutation.rb
def permutation(string1, string2)
  if string1.length != string2.length
    return false
  end

  letters_hash = {}

  string1.each_char { |char|
    if letters_hash[char] == 1
      letters_hash[char] += 1
    else
      letters_hash[char] = 1
    end
  }

  string2.each_char {|char|
    if letters_hash[string2[char]] == nil || letters_hash[string2[char]] < 0
      return false
    else
      letters_hash[string2[char]] -= 1
    end

  }
  return true
end

string1 = "robin"
string2 = "nibor"

p permutation(string1, string2)
